src/foundry/client-esm/dice/roll.d.mts

```diff
@@ -0,0 +1,636 @@
+   *                  (default: `{}`)
+   */
+  protected _evaluateSync(options?: InexactPartial<Roll.Options>): Roll.Evaluated<this>;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Evaluate an AST synchronously.
+   * @param node    - The root node or term.
+   * @param options - Options which inform how evaluation is performed
+   *                  (default: `{}`)
+   */
+  protected _evaluateASTSync(node: RollParseNode | RollTerm, options?: InexactPartial<Roll.Options>): string | number;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Safely evaluate the final total result for the Roll using its component terms.
+   * @returns The evaluated total
+   */
+  protected _evaluateTotal(): number;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Alias for evaluate.
+   * @see Roll#evaluate
+   * @param options - Options passed to Roll#evaluate.
+   */
+  roll(options?: InexactPartial<Roll.Options>): Roll.Evaluated<this> | Promise<Roll.Evaluated<this>>;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Create a new Roll object using the original provided formula and data.
+   * Each roll is immutable, so this method returns a new Roll instance using the same data.
+   * @param options - Evaluation options passed to Roll#evaluate
+   * @returns A new Roll object, rolled using the same formula and data
+   */
+  reroll(options?: InexactPartial<Roll.Options>): Roll.Evaluated<this> | Promise<Roll.Evaluated<this>>;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Recompile the formula string that represents this Roll instance from its component terms.
+   * @returns The re-compiled formula
+   */
+  resetFormula(): string;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Propagate flavor text across all terms that do not have any.
+   * @param flavor -The flavor text.
+   */
+  propagateFlavor(flavor: string): void;
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  toString(): string;
+
+  /* -------------------------------------------- */
+  /*  Static Class Methods                        */
+  /* -------------------------------------------- */
+
+  /**
+   * A factory method which constructs a Roll instance using the default configured Roll class.
+   * @typeParam D - the type of data object against which to parse attributes within the formula
+   * @param formula - The formula used to create the Roll instance
+   * @param data    - The data object which provides component data for the formula
+   * @param options - Additional options which modify or describe this Roll
+   * @returns The constructed Roll instance
+   */
+  static create<D extends Record<string, unknown> = {}>(
+    formula: string,
+    data?: D,
+    options?: InexactPartial<Roll.Options>,
+  ): typeof CONFIG.Dice.rolls extends [infer T] ? T : Roll<D>; // TODO: Unable to find config
```

Comment:
It seems like whatever CONFIG.Dice.rolls` should still be passed `T`. I guess maybe this is a part of the TODO?